src/basic/process-util.c

```diff
@@ -388,7 +388,7 @@ int is_kernel_thread(pid_t pid) {
         bool eof;
         FILE *f;
 
-        if (pid == 0 || pid == 1) /* pid 1, and we ourselves certainly aren't a kernel thread */
+        if (pid == 0 || pid == 1 || pid == getpid()) /* pid 1, and we ourselves certainly aren't a kernel thread */
```

Comment:
yupp, this PR predates the getpid_cached() PR, hence it's based on a version without it. Will rebase.